src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -1490,530 +71,116 @@ internal static unsafe uint IUnknown_Release(IntPtr pThis)
-            vftbl[3] = (IntPtr)(delegate* unmanaged<IntPtr, uint, int>)&ComWrappers.IReferenceTrackerHost_DisconnectUnusedReferenceSources;
-            vftbl[4] = (IntPtr)(delegate* unmanaged<IntPtr, int>)&ComWrappers.IReferenceTrackerHost_ReleaseDisconnectedReferenceSources;
-            vftbl[5] = (IntPtr)(delegate* unmanaged<IntPtr, int>)&ComWrappers.IReferenceTrackerHost_NotifyEndOfReferenceTrackingOnThread;
-            vftbl[6] = (IntPtr)(delegate* unmanaged<IntPtr, IntPtr, IntPtr*, int>)&ComWrappers.IReferenceTrackerHost_GetTrackerTarget;
-            vftbl[7] = (IntPtr)(delegate* unmanaged<IntPtr, long, int>)&ComWrappers.IReferenceTrackerHost_AddMemoryPressure;
-            vftbl[8] = (IntPtr)(delegate* unmanaged<IntPtr, long, int>)&ComWrappers.IReferenceTrackerHost_RemoveMemoryPressure;
-            return (IntPtr)vftbl;
-        }
-
-        private static IntPtr GetContextToken()
-        {
-#if TARGET_WINDOWS
-            Interop.Ole32.CoGetContextToken(out IntPtr contextToken);
-            return contextToken;
-#else
-            return IntPtr.Zero;
-#endif
-        }
-
-        // Wrapper for IWeakReference
-        private static unsafe class IWeakReference
-        {
-            public static int Resolve(IntPtr pThis, Guid guid, out IntPtr inspectable)
-            {
-                fixed (IntPtr* inspectablePtr = &inspectable)
-                    return (*(delegate* unmanaged<IntPtr, Guid*, IntPtr*, int>**)pThis)[3](pThis, &guid, inspectablePtr);
-            }
-        }
+        internal static unsafe IntPtr DefaultIUnknownVftblPtr => (IntPtr)Unsafe.AsPointer(in VtableImplementations.IUnknown);
+        internal static unsafe IntPtr TaggedImplVftblPtr => (IntPtr)Unsafe.AsPointer(in VtableImplementations.ITaggedImpl);
+        internal static unsafe IntPtr DefaultIReferenceTrackerTargetVftblPtr => (IntPtr)Unsafe.AsPointer(in VtableImplementations.IReferenceTrackerTarget);
 
-        // Wrapper for IWeakReferenceSource
-        private static unsafe class IWeakReferenceSource
+        private static class VtableImplementations
         {
-            public static int GetWeakReference(IntPtr pThis, out IntPtr weakReference)
+            public unsafe struct IUnknownVftbl
             {
-                fixed (IntPtr* weakReferencePtr = &weakReference)
-                    return (*(delegate* unmanaged<IntPtr, IntPtr*, int>**)pThis)[3](pThis, weakReferencePtr);
+                public delegate* unmanaged<IntPtr, Guid*, IntPtr*, int> QueryInterface;
+                public delegate* unmanaged<IntPtr, int> AddRef;
+                public delegate* unmanaged<IntPtr, uint> Release;
             }
-        }
 
-        private static object? ComWeakRefToObject(IntPtr pComWeakRef, long wrapperId)
-        {
-            if (wrapperId == 0)
+            public unsafe struct IReferenceTrackerTargetVftbl
             {
-                return null;
+                public IUnknownVftbl IUnknownVftbl;
```

Comment:
This will break the ILC optimization, as it only kicks in for "vtable-like" types, which must have only function pointer fields. You should just repeat the `IUnknown` vtable slot fields instead of reusing `IUnknownVftbl` as the field type.